### DISCUSSION TITLE
Is babel-plugin-react-native-web still required with Next.js 12?

### DISCUSSION THREAD

**Discussion by steida** [necolas/react-native-web#2217] [Q&A] [answered]
Next.js 12 has a new Rust compiler that does not support Babel.
From the babel-plugin-react-native-web readme:

`A Babel plugin that will alias react-native to react-native-web and exclude any modules not required by your app (keeping bundle size down).
`

There is already aliasing in Webpack and as I manually checked, babel-plugin-react-native-web actually makes the build slightly bigger (around 4 kB).

So my question is, is babel-plugin-react-native-web still recommended?

**Comment by danbim** [comment]
@steida Any revelations on this one? :)

**Comment by danbim** [comment]
Btw... here's the relevant documentation: https://nextjs.org/docs/advanced-features/compiler#unsupported-features

**Comment by nissy-dev** [comment, marked as answer]
I think the babel-plugin-react-native-web is not necessary in many cases in the term of tree shaking or keeping bundle size down.

Actually, I confirmed removing all the unused code from react-native-web about [the next.js example](https://github.com/vercel/next.js/tree/canary/examples/with-react-native-web) without using babel-plugin-react-native-web (only using [the package aliasing](https://necolas.github.io/react-native-web/docs/setup/#package-aliasing)).  The total module size is 70.3kB, but the bundled size is only 14kB.

<img width="1437" alt="SS_ 2022-12-10 19 19 41" src="https://user-images.githubusercontent.com/17228098/206852129-87478f92-f709-4fb1-8d28-4516a542d2ad.png">

bundlephobia also said that react-native-web package is tree shakeable.
https://bundlephobia.com/package/react-native-web@0.18.10

**Reply by danbim** [reply]
Thanks @nissy-dev ‚ Sounds promising :). Given that you’ve already done this work, could you share how your next.js configuration looks like to work with react-native-web?

**Reply by nissy-dev** [reply]
The configuration is the same as https://github.com/vercel/next.js/blob/canary/examples/with-react-native-web/next.config.js
